primefaces/src/main/java/org/primefaces/component/selectmanycheckbox/SelectManyCheckboxRenderer.java

```diff
@@ -82,9 +82,6 @@ protected void encodeMarkup(FacesContext context, SelectManyCheckbox checkbox) t
             encodeCustomLayout(context, checkbox);
             writer.endElement("span");
         }
-        else if ("grid".equals(layout)) {
-            encodeLegacyTabularLayout(context, checkbox, layout);
```

Comment:
Maybe keep the else if and throw an exception?

Reply:
good idea